pkg/vsphere/simulator/virtual_machine.go

```diff
@@ -129,39 +138,88 @@ func (vm *VirtualMachine) configure(spec *types.VirtualMachineConfigSpec) types.
 	return nil
 }
 
-func (vm *VirtualMachine) create(spec *types.VirtualMachineConfigSpec) types.BaseMethodFault {
-	p, fault := parseDatastorePath(spec.Files.VmPathName)
+func (vm *VirtualMachine) useDatastore(name string) *Datastore {
+	host := Map.Get(*vm.Runtime.Host).(*HostSystem)
+
+	ds := Map.FindByName(name, host.Datastore).(*Datastore)
+
+	if Map.FindByName(name, vm.Datastore) == nil {
+		vm.Datastore = append(vm.Datastore, ds.Reference())
+	}
+
+	return ds
+}
+
+func (vm *VirtualMachine) setLog(w io.Writer) {
+	vm.log = log.New(w, "vmx ", log.Flags())
+}
+
+func (vm *VirtualMachine) createFile(spec string, name string) (*os.File, types.BaseMethodFault) {
+	p, fault := parseDatastorePath(spec)
 	if fault != nil {
-		return fault
+		return nil, fault
 	}
 
-	host := Map.Get(*vm.Runtime.Host).(*HostSystem)
-	ds := Map.FindByName(p.Datastore, host.Datastore).(*Datastore)
+	ds := vm.useDatastore(p.Datastore)
+
+	file := path.Join(ds.Info.GetDatastoreInfo().Url, p.Path)
+
+	if name != "" {
+		if path.Ext(file) != "" {
+			file = path.Dir(file)
+		}
+
+		file = path.Join(file, name)
+	}
 
-	// Assuming local datastore for now.  TODO: should be via datastore browser / file manager
-	dir := ds.Info.GetDatastoreInfo().Url
+	dir := path.Dir(file)
 
-	vmx := path.Join(dir, p.Path)
+	_ = os.MkdirAll(dir, 0700)
```

Comment:
@dougm Perhaps we make this configurable - I agree that we should try to avoid the rename at all, but it's not completely within our control. Having the simulator deliberately from known behavior won't help us ensure that we correctly handle the case where rename does happen.